datanode/api/trading_data_v2.go

```diff
@@ -3048,6 +3055,78 @@ func (t *tradingDataServiceV2) GetVegaTime(ctx context.Context, _ *v2.GetVegaTim
 
 // -- NetworkHistory --.
 
+func (t *tradingDataServiceV2) ExportNetworkHistory(req *v2.ExportNetworkHistoryRequest, stream v2.TradingDataService_ExportNetworkHistoryServer) error {
+	defer metrics.StartAPIRequestAndTimeGRPC("ExportNetworkHistory")()
+
+	if req.TableName == "" {
+		return formatE(ErrNetworkHistoryNoTableName, errors.New("empty table name"))
+	}
+
+	segments, err := t.networkHistoryService.ListAllHistorySegments()
+	if err != nil {
+		return formatE(ErrListAllNetworkHistorySegment, err)
+	}
+
+	ch, err := networkhistory.GetContiguousHistory(segments, req.FromBlock, req.ToBlock)
+	if err != nil {
+		return formatE(ErrNetworkHistoryGetContiguousSegments, err)
+	}
+
+	header := metadata.Pairs("Content-Disposition", fmt.Sprintf("attachment;filename=vega-%s-%06d-%06d.csv", req.TableName, ch.HeightFrom, ch.HeightTo))
+	if err := stream.SendHeader(header); err != nil {
+		return formatE(ErrSendingGRPCHeader, err)
+	}
+
+	var chunk bytes.Buffer
+	rowNum := 1
+
+	for _, segment := range ch.SegmentsOldestFirst {
+		outerTarFileReader, err := t.networkHistoryService.GetHistorySegmentReader(stream.Context(), segment.GetHistorySegmentId())
+		if err != nil {
+			return formatE(ErrNetworkHistoryOpeningSegment, err)
+		}
+
+		segmentData, err := fsutil.ReadNetworkHistorySegmentData(outerTarFileReader, req.TableName)
+		if err != nil {
+			return formatE(ErrNetworkHistoryExtractingSegment, err)
+		}
+
+		scanner := bufio.NewScanner(segmentData)
+		for ; scanner.Scan(); rowNum++ {
+			row := scanner.Text()
+			row = strings.Replace(row, "\t", ",", -1) // postgres dump file separated by tabs
```

Comment:
Would it be better to switch the postgres format to csv?,  it may be better to do this in case there is any formatting going on that we are not aware of, that way an exported csv will be guaranteed to be importable to postgres for a given schema version, this could be very useful,  just need to do a before and after test on segment size and load time to check no meaningful impact.

Reply:
Yes probably - it addresses a number of problems actually;
- you can include headers e.g. `COPY assets TO stdout WITH (FORMAT csv, HEADER);`
- I think what I have at the moment will break if, for example, there's a comma in a text field - it needs escaping, then you need to escape the escape character if that's in a field too. 

I'm more than happy to change it to do that but it means then that network history files won't be compatible over that change unless we add some check or something when importing

Reply:
Yeah, as agreed on slack, if we do it now, no need to worry about backwards compatibility.